code/controller/actuator.cpp

```diff
@@ -32,11 +32,11 @@ char* const Actuator::convertDataToBytes(long int data) {
         data = intPow(BYTE_RANGE, 4) + data;
     }
 
-    char result[DATA_SIZE];
+    static char result[DATA_SIZE];

```

Comment:
why did you make this static? This may lead to headaches down the road unless you made the change for a good reason.